iroh-resolver/src/content_loader.rs

```diff
@@ -161,11 +161,21 @@ impl FullLoader {
     async fn fetch_gateway(&self, cid: &Cid) -> Result<Option<LoadedCid>> {
         match self.next_gateway().await {
             Some(url) => {
-                let data = reqwest::get(url.as_url(cid)).await?.bytes().await?;
-                Ok(Some(LoadedCid {
-                    data,
-                    source: Source::Http(url.as_string()),
-                }))
+                let response = reqwest::get(url.as_url(cid)).await?;
+                // Filter out non http 200 responses.
+                if !response.status().is_success() {
+                    return Ok(None);
+                }
+                let data = response.bytes().await?;
+                // Make sure the content is not tampered with.
+                if iroh_util::verify_hash(cid, &data) == Some(true) {
+                    Ok(Some(LoadedCid {
+                        data,
+                        source: Source::Http(url.as_string()),
+                    }))
+                } else {
+                    Ok(None)
```

Comment:
this should return an error

Reply:
ok - but why don't we also return an error when there is no gateway?

Reply:
that’s what the None is for in me head, because we didn’t fail to retrieve, we just didn’t have one configured

Reply:
I swiched to errors in the latest commit.